### DISCUSSION TITLE
How robust is nats.ws?

### DISCUSSION THREAD

**Discussion by idc77** [nats-io/nats.ws#147] [Q&A]
I've been doing limited exploring of options to notify clients of new messages (in a dating "app")
In the past I've used browser js WebSocket but have not bothered to do any failsafety, like reconnection of interrupted connection or similar.
I've also tried sse (EventSource) with very bad results.

How robust is nats.ws with Jetstream?

I expect my dating app to be used mostly on mobile devices, where connections are potentially ever changing.
Would nats Jetstream with nats.ws be a good fit? If not, do you have alternative suggestions?

**Comment by aricart** [comment]
The main question is whether a w3c compliant socket is available in your environment.  The websocket transport has full support for JetStream.
